### PR TITLE
HPCC-10709 Allow dafilesrv throttle limit to be configured

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday @richardkchapman - please review

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: Fail
HPCC Stop: OK

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -4280,6 +4330,64 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
...
+                     * but still need to consume the signals, to avoid issues if throttling changed again
+                     */
+                    throttleLimit = 0;
+                }
+
+                // NB: This is not expected to take long
+                CCycleTimer timer;
+                while (diff)
+                {
+                    if (throttlesem.wait(1000))
```
Comment (ghalliday) [inline common/remote/sockfile.cpp]: I'm not sure about this line.  If the throttle limit is set to 0, and there are several items waiting, could this grab all the signals, and leave those items forever waiting?

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

**Comment by jakesmith** [comment]
@ghalliday - can you take another look at the approach taken to disabling/re-enabling throttling..

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

File: common/remote/sockfile.cpp [hunk 2]
```diff
@@ -4280,6 +4331,77 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
...
+                if (disabledThrottleLimit) // if transitioning from disabled to some throttling
+                {
+                    assertex(0 == throttleLimit);
+                    delta = _throttleLimit - disabledThrottleLimit; // + or -
+                    disabledThrottleLimit = 0;
+                }
+                else
+                    delta = _throttleLimit - throttleLimit; // + or -
+            }
+            else
```
Comment (ghalliday) [inline common/remote/sockfile.cpp]: else if (disabledThrottleLimit != 0)

otherwise setting to 0 2 times in a row will lose the value.

**Comment by ghalliday** [comment]
@jakesmith one minor issue, looks good otherwise.

File: common/remote/sockfile.cpp (hunk 2 above)
Comment (jakesmith) [inline common/remote/sockfile.cpp]: Right, but you mean:
```
else if (disabledThrottleLimit == 0)
```

**Comment by jakesmith** [comment]
@ghalliday - made change re. last comment, and squashed commits.

**Comment by ghalliday** [comment]
@richardkchapman looks good - please double check

**Pull request merged**